nni/compression/experiment/config/utils.py

```diff
@@ -0,0 +1,152 @@
+        if module_type in _module_types:
+            module_names_summery.add(module_name)
+        _all_module_names.add(module_name)
+
+    # add module names to summery
+    if module_names:
+        for module_name in module_names:
+            if module_name not in _all_module_names:
+                # need warning, module_name not exist
+                continue
+            else:
+                module_names_summery.add(module_name)
+
+    # remove module names in exclude_module_names from module_names_summery
+    if exclude_module_names:
+        for module_name in exclude_module_names:
+            if module_name not in _all_module_names:
+                # need warning, module_name not exist
+                continue
+            if module_name in module_names_summery:
+                module_names_summery.remove(module_name)
+
+    return list(module_names_summery)
+
+def cc_cv2ss(config: CompressionConfig, vessel: CompressionVessel):
+    search_space = {}
+    model, _, evaluator, dummy_input, _, _, _, _ = vessel.export()
+    flops, params, results = count_flops_params(model, dummy_input, verbose=False, mode='full')
+    metric = evaluator(model)
+
+    module_names_summery = _summery_module_names(model, config.module_types, config.module_names, config.exclude_module_names)
+    for module_name in module_names_summery:
+        search_space['{}{}'.format(KEY_MODULE_NAME, module_name)] = {'_type': 'uniform', '_value': [0, 1]}
+
+    assert not config.pruners or not config.quantizers
+    # hard code for test, need remove
+    search_space[KEY_PRUNERS] = {'_type': 'choice', '_value': [pruner_config.json() for pruner_config in config.pruners]}
+
+    original_target = {'flops': flops, 'params': params, 'metric': metric, 'results': results}
+
+    flops_theta = _flops_theta_helper(config.flops, flops)
+    params_theta = _flops_theta_helper(config.params, params)
+    metric_theta = _metric_theta_helper(config.metric, metric)
+    thetas = {'flops': flops_theta, 'params': params_theta, 'metric': metric_theta}
+
+    search_space[KEY_VESSEL] = {'_type': 'choice', '_value': [vessel.json()]}
+    search_space[KEY_ORIGINAL_TARGET] = {'_type': 'choice', '_value': [original_target]}
+    search_space[KEY_THETAS] = {'_type': 'choice', '_value': [thetas]}
+    return search_space
+
+def parse_params(kwargs: Dict[str, Any]):
+    pruner_config, vessel, original_target, thetas = None, None, None, None
+    config_list = []
+
+    for key, value in kwargs.items():
+        if key.startswith(KEY_MODULE_NAME):
+            config_list.append({'op_names': [key.split(KEY_MODULE_NAME)[1]], 'sparsity_per_layer': float(value)})
+        elif key == KEY_PRUNERS:
+            pruner_config = value
+        elif key == KEY_VESSEL:
+            vessel = CompressionVessel(**value)
+        elif key == KEY_ORIGINAL_TARGET:
+            original_target = value
+        elif key == KEY_THETAS:
+            thetas = value
+
```

Comment:
Should there be a warning if the key is not anyone above?

Reply:
fixed